Submitty/Submitty
# [Refactor:Router] New URL for ElectronicGraderController

### URL Changes

This controller is huge... Should split it some time after.

Most of them are Ajax endpoints. Non-ajax URLs are marked.

Rubric:
- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/rubric`

    Component-Related:
    - GET `/{_semester}/{_course}/gradeable/{gradeable_id}/components`
    - POST `/{_semester}/{_course}/gradeable/{gradeable_id}/components/verify`
    - POST `/{_semester}/{_course}/gradeable/{gradeable_id}/components/save`
    - POST `/{_semester}/{_course}/gradeable/{gradeable_id}/components/order`
    - POST `/{_semester}/{_course}/gradeable/{gradeable_id}/components/save_pages`
    - POST `/{_semester}/{_course}/gradeable/{gradeable_id}/components/new`
    - POST `/{_semester}/{_course}/gradeable/{gradeable_id}/components/delete`

    Mark-Related:
    - POST `/{_semester}/{_course}/gradeable/{gradeable_id}/components/marks/save`
    - POST `/{_semester}/{_course}/gradeable/{gradeable_id}/components/marks/save_order`
    - POST `/{_semester}/{_course}/gradeable/{gradeable_id}/components/marks/add`
    - POST `/{_semester}/{_course}/gradeable/{gradeable_id}/components/marks/delete`
    - POST `/{_semester}/{_course}/gradeable/{gradeable_id}/components/marks/stats`

Team-Related:
- POST `/{_semester}/{_course}/gradeable/{gradeable_id}/grading/teams/import`
- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/grading/teams/export`
- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/grading/teams/randomize_rotating`
- POST `/{_semester}/{_course}/gradeable/{gradeable_id}/grading/teams/new`

Actual Grading:
- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/grading/status` (non-Ajax)
- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/grading/details` (non-Ajax)
- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/grading/grade` (non-Ajax)
- POST `/{_semester}/{_course}/gradeable/{gradeable_id}/grading/comments`
- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/grading/comments`
- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/grading/graded_gradeable`
- POST `/{_semester}/{_course}/gradeable/{gradeable_id}/grading/graded_gradeable/graded_component`
- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/grading/graded_gradeable/graded_component`

Student Output:

- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/grading/student_output`
- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/grading/student_output/remove` (space removed version of student output)

### Testing
Manually tested the following:
- Go to the grading page.
- Grade and add comments. See if grades are saved.
- Edit rubric, add marks and reorder them, save.
- See student output, visualize white spaces
- Team operations: export teams, create team for students, import teams to a new gradeable
